### PR TITLE
Firestore Connecting

### PR DESCRIPTION
**Pull request by andrwnv**

### CODE REVIEW

File: api/FirestoreAPI.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+
+    constructor() {
+        if ( !firebase.apps.length ) { 
+            firebase.initializeApp(firebaseConfig); 
+            this.state.isConnected = true; 
+
+            this._firestore = firebase.firestore(); 
+            
+            let a = encrypt('+79991774634');
```
Comment (andrwnv) [inline api/FirestoreAPI.ts]: delete.

File: api/FirestoreAPI.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                                                                     eysColor:  docData.eysColor, 
+                                                                     skinColor: docData.skinColor } });
+                });
+            })
+            .catch((err: any) => { 
+                console.error('[fireStoreAPT] -> Error: cant get document: ' + err);
+                return [];
+            });
+        
+        console.log(res);
```
Comment (andrwnv) [inline api/FirestoreAPI.ts]: delete.

File: api/FirestoreAPI.ts [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                const snapData = snapshot.data();
+
+                if (snapData === undefined)
+                    return undefined;
+
+                return { hairColor: snapData.hairColor,
+                         eysColor:  snapData.eysColor,
+                         skinColor: snapData.skinColor }
+            } else {
+                console.warn('[fireStoreAPT] -> Warn: Document no exists')
```
Comment (andrwnv) [inline api/FirestoreAPI.ts]: ;

File: api/FirestoreAPI.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+                return true;
+            }
+        })
+        .catch((err: any) => {
+            console.error('[fireStoreAPT] -> Error: cant get document: ' + err);
+            return false;
+        });
+    }
+}
```
Comment (andrwnv) [inline api/FirestoreAPI.ts]: add new_line

File: api/FirestoreAPI.ts [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            
+            let a = encrypt('+79991774634');
+            a = encrypt('+79991774634');
+            console.log(decrypt(a));
+        } 
+    }
+    
+    public isConnected = (): boolean => { return this.state.isConnected; }
+
+     getUsers = async (): Promise<UserData[]> => {
```
Comment (Diklofozzz1) [inline api/FirestoreAPI.ts]: user Data | Undefined

File: api/FirestoreAPI.ts [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+        if ( !this.state.isConnected ) {
+            return [];
```
Comment (Diklofozzz1) [inline api/FirestoreAPI.ts]: user Data | Undefined

File: api/FirestoreAPI.ts [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    }; 
+
+    public setUserFields = async (userName: string, newProps: Object): Promise<boolean | undefined> => {
+        if ( !this.state.isConnected ) {
+            return false;
+        }
+
+        let userPropsRef = this._firestore.collection(this._defaultCollectionName).doc(encrypt(userName));
+        
+        const isUserExist = await userPropsRef.get().then((snapshot) => {
```
Comment (Diklofozzz1) [inline api/FirestoreAPI.ts]: Same code

Comment (Diklofozzz1) [reply api/FirestoreAPI.ts]: segregate to single function

File: api/FirestoreAPI.ts [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        }
+
+        for (let [key, value] of Object.entries(newProps)) {
+            userPropsRef.set( { [key]: value }, {merge: true} );
+        }
+
+        return true;
+    };
+
+    public createUser = async (userName: string): Promise<boolean> => {
```
Comment (Diklofozzz1) [inline api/FirestoreAPI.ts]: Undefined user

File: api/FirestoreAPI.ts [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+        if ( !this.state.isConnected ) {
+            return false;
+        }
+
+        const userPropsRef = this._firestore.collection(this._defaultCollectionName).doc(encrypt(userName));
+        
+        return await userPropsRef.get().then((snapshot) => {
```
Comment (Diklofozzz1) [inline api/FirestoreAPI.ts:171]: same code

**Pull request merged**